### PR TITLE
Feature/backend unit tests

### LINKED ISSUE #29: Backend tests
Create unit tests to test backend api.
- Test routing
- Test mongodb connection
- Test save values to mongodb database
- Test fetch data from mongodb database

### PR DESCRIPTION
**Pull request by jjsoong**
Closes #29.
Added backend unit tests to project.
Reformatted some of the backend (specifically server.js) to implement the tests.

### CODE REVIEW

**Review by hc-rabid — APPROVED** [review]

**Review by Justinzoorob — APPROVED** [review]

File: server/server.js [hunk 1]
```diff
@@ -1,38 +1,82 @@
...
+    //         });
+    // }
+
+    server.drop = async function() {
+        await FormPost.deleteMany();
+    };
+
+    server.close = function() {
+        mongoose.connection.close(false);
+    }
```
Comment (khalidkanaan) [inline server/server.js]: add ;

File: server/tests/server.test.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            message: "Hello, world!"
+        };
+        
+        request(server.app)
+            .post('/api/save')
+            .send(testInput)
+            .set('Accept', 'application/json')
+            .expect('Content-Type', 'application/json; charset=utf-8')
+            .expect(200, { msg: 'We received your data!' }, done);
+            // Does not check if data is added to db properly yet.
```
Comment (khalidkanaan) [inline server/tests/server.test.js]: delete commented code.

File: server/tests/server.test.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+            .send({
+                name: "Jane Doe",
+                level: 20,
+                rank: "B+",
+                isGood: false
+            })
+            .set('Accept', 'application/json')
+            .expect('Content-Type', 'application/json; charset=utf-8')
+            .expect(400, { msg: 'Failed to save your data.' }, done);
+            // Does not check if data is not added to db properly yet.
```
Comment (khalidkanaan) [inline server/tests/server.test.js]: delete commented code.

File: server/tests/server.test.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        await server.drop();
+    });
+
+    afterAll(async () => {
+        await server.drop();
+        server.close();
+    });
+});
+
+// Function planned for after-POST testing, but put on hold for now.
```
Comment (khalidkanaan) [inline server/tests/server.test.js]: delete commented code.

File: server/server.js [hunk 5]
```diff
@@ -1,38 +1,82 @@
...
+    server.add = function (data) {
+        const newData = new FormPost(data);
+        newData.save((err) => {
+            if (err) {
+                console.error(`Could not save data to mongodb.\nError: ${err}`);
+            }
+        });
+    };
+
+    // server.get = function() {
```
Comment (khalidkanaan) [inline server/server.js]: delete commented code.

**Review by mputnik — APPROVED** [review]

**Review by Dweevil — COMMENTED** [review]
looks good

**Review by khalidkanaan — APPROVED** [review]

**Review by mputnik — APPROVED** [review]

File: server/server.js (hunk 5 above)
Comment (jjsoong) [reply server/server.js]: Resolved with commit f5897fe7de72865b94ed3e75f5ddee7fd31a3f63

File: server/tests/server.test.js (hunk 4 above)
Comment (jjsoong) [reply server/tests/server.test.js]: Resolved with commit f5897fe7de72865b94ed3e75f5ddee7fd31a3f63

File: server/tests/server.test.js (hunk 3 above)
Comment (jjsoong) [reply server/tests/server.test.js]: Resolved with commit f5897fe7de72865b94ed3e75f5ddee7fd31a3f63

File: server/tests/server.test.js (hunk 2 above)
Comment (jjsoong) [reply server/tests/server.test.js]: Resolved with commit f5897fe7de72865b94ed3e75f5ddee7fd31a3f63

File: server/server.js (hunk 1 above)
Comment (jjsoong) [reply server/server.js]: Resolved with commit 8f8939661c654135c0f4b77c2dab060f44b14ee9

**Review by Justinzoorob — APPROVED** [review]

**Pull request merged**